server/src/main/java/org/opensearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -412,8 +416,9 @@ protected void handlePrimaryRequest(final ConcreteShardRequest<Request> request,
             releasable::close
         );
 
-        try {
-            new AsyncPrimaryAction(request, listener, (ReplicationTask) task).run();
+        final Span span = getTracer().startSpan(SpanBuilder.from("shardPrimaryWrite", clusterService.localNode().getId(), request.getRequest().shardId()));
```

Comment:
Should we also add the index as an attribute?

Reply:
Index name is being fetched from the Shard Id itself.